### ISSUE TITLE
java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.String

### ISSUE THREAD

**Issue by asimplepeople** [apache/dubbo#11157]
- [ ] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 2.7.3
* Operating System version: xxx
* Java version: xxx

### Steps to reproduce this issue

1. 编写request

public class xxxxxxRequest implements Serializable {
    private static final long serialVersionUID = 1L;
    private List<String> ids;
}
泛化调用
"request": {
      "ids": [1]
    }
2. ![image](https://user-images.githubusercontent.com/26264536/207911883-73482c62-9987-45c7-98fe-5875de59ad90.png)
4. ![image](https://user-images.githubusercontent.com/26264536/207912387-a73c4ab2-480c-4062-ab7b-1aeb920b6a6f.png)

Pls. provide [GitHub address] to reproduce this issue.

### Expected Behavior
正确的抛出类型转换错误而不是在调用for循环的时候抛类型转换错误

### Actual Behavior

If there is an exception, please attach the exception trace:

```
Just put your stack trace here!
```

**Comment by EarthChen** [comment]
plz update latest version to try it.

**Comment by EarthChen** [comment]
plz update latest version to try it.

**Comment by samyScott** [comment]
please assign to me

**Comment by CrazyHZM** [comment]
@samyScott 
Is there any progerss?

**Comment by ghostly72** [comment]
hi, 
I added a pr, trying to resolve this. Can u take a look at it? @CrazyHZM 
PojoUtils.realize() ignored the generic parameter when deserializing Collection fields. As a result, it always converted elements as Object instead of using the declared element type from ParameterizedType. So i added an else block to correct generic element type from ParameterizedType and throw an early IllegalArgumentException if type conversion fails (instead of deferring it to user code). 

And tested it using a unit test, mimicking this situation and got an illegal argument error as expected in the deserialization stage

happy to hear suggestions and improvements
thanks

**Comment by ghostly72** [comment]
hi,
I added a check for early type-validation inside `PojoUtils.realize()` to fix the case where List<String> fields were being deserialized as List<Integer>, leading to ClassCastException at runtime. The fix works for that specific case (now throws illegalargumentexception early), but it broke existing tests like:
```
Map<String, Object> map = PojoUtils.generalize(person);
PojoUtils.realize(map, Person.class);

```
here the framework intentionally converts a person to/from a map. So when Dubbo tries to realize a person object into a hashmap(or vice versa), my check says: “person is not a hashmap, so throw exception”.

suggestions on how best to handle this without breaking existing behavior? also if this approach fundamentally conflicts with Dubbo’s design, I'll close this PR, do let me know

**Comment by zrlw** [comment]
> conflicts with Dubbo’s design

yes, ```Map``` is designed as a generalized type for any java Object.
